src/cryptography/hazmat/backends/openssl/backend.py

```diff
@@ -1524,6 +1501,30 @@ def create_x509_crl(self, builder, private_key, algorithm):
 
         return _CertificateRevocationList(self, x509_crl)
 
+    def _create_x509_extensions(self, extensions, handlers):
```

Comment:
I'm not sure I follow. In the case of the CSR we need to create the extensions, push them onto a stack, then add that stack to the CSR. For the cert case we can just add the extensions directly to the X509 pointer. I could pass in the stack/x509 pointer and an add function, but the add functions have different arguments anyway.